Fix Mini Pets

Oroxzy opened the pull request:
**Fixed some values and let them use their auras.**
![88641-mini-diablo](https://user-images.githubusercontent.com/357606/210226168-00d8f5dc-ce76-4b5b-9f17-12f1667254f0.jpg)

fixes #239 

**Proof:**
use this query on mass parse db with an existing vmangos world db named "mangos"
```sql
SELECT
    spell_template.entry AS spell_entry,
    spell_template.name AS spell_name,
    creature_template.entry,
    creature_template2.name AS creature_name,
	creature_template.auras,
	CAST(IFNULL(creature_template.auras, 0) AS int),
	CAST(IFNULL(creature_template2.auras, 0) AS int),
    CONCAT('UPDATE `creature_template` SET ',
    IF ((CAST(IFNULL(creature_template.auras, 0) AS int) = CAST(IFNULL(creature_template2.auras, 0) AS int)), '', CONCAT('`auras`=', IFNULL(creature_template.auras, 0))),
    IF ((creature_template.unit_flags = creature_template2.unit_flags), '', CONCAT(', `unit_flags`=', creature_template.unit_flags)),
    IF ((creature_template.speed_walk = creature_template2.speed_walk), '', CONCAT(', `speed_walk`=', creature_template.speed_walk)),
    IF ((creature_template.speed_run = creature_template2.speed_run), '', CONCAT(', `speed_run`=', creature_template.speed_run)),
    IF ((creature_template.base_attack_time = creature_template2.base_attack_time), '', CONCAT(', `base_attack_time`=', creature_template.base_attack_time)),
    IF ((creature_template.ranged_attack_time = creature_template2.ranged_attack_time), '', CONCAT(', `ranged_attack_time`=', creature_template.ranged_attack_time)),
    IF ((creature_template.faction = creature_template2.faction), '', CONCAT(', `faction`=', creature_template.faction)),
    ' WHERE `entry`=', creature_template.entry, ';') AS query
FROM creature_template
JOIN mangos.spell_template ON creature_template.entry = mangos.spell_template.effectMiscValue1
JOIN mangos.creature_template AS creature_template2 ON creature_template.entry = creature_template2.entry
WHERE
    creature_template.entry IN (SELECT effectMiscValue1 FROM mangos.spell_template WHERE effect1 = 97 OR effect2 = 97 OR effect3 = 97 GROUP BY effectMiscValue1) AND (
    NOT creature_template.speed_walk = creature_template2.speed_walk OR
    NOT creature_template.speed_run = creature_template2.speed_run OR
    NOT creature_template.faction = creature_template2.faction OR
    NOT creature_template.base_attack_time = creature_template2.base_attack_time OR
    NOT creature_template.ranged_attack_time = creature_template2.ranged_attack_time OR
    NOT creature_template.unit_flags = creature_template2.unit_flags OR
    NOT CAST(IFNULL(creature_template.auras, 0) AS int) = CAST(IFNULL(creature_template2.auras, 0) AS int)) AND (
    spell_template.effect1 = 97 OR
    spell_template.effect1 = 97 OR
    spell_template.effect1 = 97 )
```

Oroxzy committed:
Fix Mini Pets

ratkosrb commented on src/game/Objects/Pet.cpp:
```diff
@@ -2287,9 +2288,22 @@ bool Pet::Create(uint32 guidlow, CreatureCreatePos& cPos, CreatureInfo const* ci
+    {
+        SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PLAYER | UNIT_FLAG_IMMUNE_TO_NPC); // always non-attackable
 
+        CreatureInfo const* normalInfo = ObjectMgr::GetCreatureTemplate(this->GetEntry());
```
you are given the cinfo as parameter, why are you looking it up

ratkosrb commented:
Why are you updating the ranged attack time? This is not even a sniffable value.

ratkosrb commented:
And the unit flags are assigned by core for mini pets, why do u set them in db too? Do these entries actually have the Immune to PC/NPC static flags in their creature difficulty record (blizzard's equivalent of template) ?

ratkosrb commented:
And what about the faction update, aren't mini pets always their owner's faction? I only checked 9657, but in sniffs its the player's faction, not 35, and in creature difficulty its 188, so where is the update to 35 coming from?

Oroxzy committed:
Update 20230102120931_world.sql

ratkosrb commented on src/game/Objects/Pet.cpp:
```diff
@@ -20,6 +20,7 @@
  */
 
 #include "Pet.h"
+#include "Creature.h"
```
Pet.h already includes Creature.h, since Pet inherits from Creature.

Oroxzy committed:
add murloc dance

Oroxzy committed:
Update Pet.cpp

Oroxzy committed:
Update Pet.cpp

Oroxzy committed:
Update SpellAuras.cpp

Oroxzy committed:
Update 20230102120931_world.sql

Oroxzy committed:
Update SpellAuras.cpp

Oroxzy committed:
Merge remote-tracking branch 'upstream/development' into mini_pets

ratkosrb committed:
Remove wrong speed updates.

Only 1 spawn is seen with that speed, and it has aura Spirit of Zandalar which increases speed... Totally wrong update!

ratkosrb committed:
Do not lookup creature info when its already given as parameter to the function.
